utility/types/vote_test.go

```diff
@@ -11,36 +11,38 @@ func TestVoteValidateBasic(t *testing.T) {
 	publicKey, err := crypto.GeneratePublicKey()
 	require.NoError(t, err)
 	testHash := crypto.SHA3Hash([]byte("fake_hash"))
-	v := LegacyVote{
-		PublicKey: publicKey.Bytes(),
-		Height:    1,
-		Round:     2,
-		Type:      DoubleSignEvidenceType,
-		BlockHash: testHash,
-	}
+	v := createLegacyVote(publicKey.Bytes(), testHash, 1, DoubleSignEvidenceType)
 	require.NoError(t, v.ValidateBasic())
 	// bad public key
-	v2 := v
-	v2.PublicKey = []byte("not_a_public_key")
+	v2 := createLegacyVote([]byte("not_a_public_key"), testHash, 1, DoubleSignEvidenceType)
 	badPkLen := len(v2.PublicKey)
 	require.Equal(t, v2.ValidateBasic(), ErrInvalidPublicKeyLen(crypto.ErrInvalidPublicKeyLen(badPkLen)))
 	// no public key
 	v2.PublicKey = nil
 	require.Equal(t, v2.ValidateBasic(), ErrEmptyPublicKey())
 	// bad hash
-	v3 := v
-	v3.BlockHash = []byte("not_a_hash")
+	v3 := createLegacyVote(publicKey.Bytes(), []byte("not_a_hash"), 1, DoubleSignEvidenceType)
 	badBlockHashLen := len(v3.BlockHash)
 	require.Equal(t, v3.ValidateBasic(), ErrInvalidHashLength(crypto.ErrInvalidHashLen(badBlockHashLen)))
 	// no hash
 	v3.BlockHash = nil
 	require.Equal(t, v3.ValidateBasic(), ErrEmptyHash())
 	// negative height
-	v4 := v
+	v4 := createLegacyVote(publicKey.Bytes(), testHash, -1, DoubleSignEvidenceType)
```

Comment:
why not do just `proto.Clone(v)`?

Reply:
With:

```go
v2 := proto.Clone(v)
v2.PublicKey = []byte("not_a_public_key")
```

I would get this error 

![image](https://user-images.githubusercontent.com/53987565/216764195-c422bc08-9219-43a3-8f30-53966a52586f.png)

I am sure there is a better way around this error but I couldn't find much help online so I went with new `*LegacyVote` structs for each test

Reply:
I didn't test this out myself, but I think the reasoning for this might be because `proto.Clone` returns a generic `proto.Message`, but `PublicKey` is a property of the `LegacyVote` proto.

What if you type cast it first: `v2 := proto.Clone(v).(LegacyVote)`?

Reply:
@Olshansk 
![image](https://user-images.githubusercontent.com/53987565/217096317-2669f5f0-1be7-4bbe-aa74-f0f4a26cfbf3.png)

I tried this I think its because `proto.Clone()` doesn't return a generic message but a reflection of one which must behave differently I tried reading the docs about it but didn't make much progress.

Reply:
```go
	v3 := proto.Clone(v).(*LegacyVote)
	v.BlockHash = []byte("not_a_hash")
```

![Screenshot 2023-02-06 at 2 05 51 PM](https://user-images.githubusercontent.com/1892194/217098339-90a73b8e-467a-4ed7-95d9-25900c70314c.png)

Reply:
My bad forgot the pointer 🤣 pushed